### PR TITLE
Add GitFiend app v0.22.8

### PR DESCRIPTION
**Pull request by untone-survive**
Add GitFiend git client v0.22.8

After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not [already refused].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask

### CODE REVIEW

File: Casks/gitfiend.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+cask 'gitfiend' do
+  version '0.22.8'
+  sha256 '8d0eaaf47074c28d97f8fca7780e11f868adb8ab97f18c8e196f54817a86237d'
+
+  url "https://gitfiend.com/resources/GitFiend-#{version}.dmg"
```
Comment (vladimyr) [inline Casks/gitfiend.rb:5]:
```suggestion
  url "https://gitfiend.com/resources/GitFiend-#{version}.dmg"
  appcast 'https://gitfiend.com/app-info'
```

File: Casks/gitfiend.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  name 'GitFiend'
+  homepage 'https://gitfiend.com/'
```
Comment (vladimyr) [inline Casks/gitfiend.rb:7]:
```suggestion
  homepage 'https://gitfiend.com/'

  auto_updates true
```

File: Casks/gitfiend.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+
+  app 'GitFiend.app'
```
Comment (vladimyr) [inline Casks/gitfiend.rb:9]:
```suggestion
  app 'GitFiend.app'

  zap trash: [
               '~/Library/Application Support/GitFiend',
               '~/Library/Preferences/com.tobysuggate.gitfiend.plist',
               '~/Library/Saved Application State/com.tobysuggate.gitfiend.savedState',
             ]
```

**Comment by vladimyr** [comment]
GitFiend supports launching from the command line using `open`: https://github.com/GitFiend/Support/issues/4#issuecomment-517892196 but unfortunately, it does not handle relative paths so here is quick n dirty wrapper that might help cross that gap:
```rb
  # shim script (https://github.com/Homebrew/homebrew-cask/issues/18809)
  shimscript = "#{staged_path}/gitfiend.wrapper.sh"
  binary shimscript, target: 'gitfiend'

  preflight do
    IO.write shimscript, <<~EOS
      #!/bin/sh

      git_dir=$(perl -e 'use Cwd "abs_path"; print abs_path(shift)' "$1")
      open -a GitFiend --args "$git_dir"
    EOS
  end
```
This Perl oneliner makes me sick but even that monstrosity is better than emulating missing `realpath` in pure `bash`.

**Comment by vitorgalvao** [comment]
Using `open -a` doesn’t qualify as a shimscript, only things inside the bundle. So this is good as is.

**Comment by vladimyr** [comment]
> Using `open -a` doesn’t qualify as a shimscript, only things inside the bundle. So this is good as is.

Binary can be run directly but it needs to be sent to the background because it will steal terminal session.

**Comment by vitorgalvao** [comment]
Any disadvantage to running it with `&` at the end?

**Comment by vladimyr** [comment]
> Any disadvantage to running it with `&` at the end?

It needs to be run using `nohup` to ensure it continues working after the terminal gets closed and with redirected stdout/err. It seemed messier than using `open -a` but I've just created PR so we can move the conversation there #81491

**Comment by vitorgalvao** [comment]
> It needs to be run using `nohup` to ensure it continues working after the terminal gets closed and with redirected stdout/err. It seemed messier than using `open -a`

I agree. Change the PR to `open -a` and I’ll merge that.

**Pull request merged**